crates/wasi/src/preview2/network.rs

```diff
@@ -53,27 +82,46 @@ impl SystemNetwork {
     pub fn new() -> Self {
         Self {}
     }
+
+    /// Non-boxing variant of [Network::resolve_addresses]
+    pub fn resolve_addresses(
+        &mut self,
+        name: String,
+    ) -> impl Future<Output = io::Result<Vec<IpAddr>>> + Send + Sync + 'static {
+        async move { resolve_addresses(&name).await }
+    }
+
+    /// Non-boxing variant of [Network::new_tcp_socket]
+    pub fn new_tcp_socket(&mut self, family: SocketAddrFamily) -> io::Result<SystemTcpSocket> {
+        SystemTcpSocket::new(family)
+    }
```

Comment:
Because the functions they call are internal to the crate. We could open those up, but I think it makes sense (at least conceptually) that consumers can't fabricate a SystemTcpSocket without going through SystemNetwork.